pkg/release/release.go

```diff
@@ -128,78 +128,105 @@ func (r *Release) Condition() ConditionSet {
 		return microerror.Mask(tillerNotFoundError)
 	} else if err != nil {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted release %#q", releaseName))
-
 	return nil
 }
 
-func (r *Release) EnsureDeleted(ctx context.Context, name string) error {
-	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
+// EnsureDeleted makes sure the release is deleted and purged and all
+// conditions are met.
+func (r *Release) EnsureDeleted(ctx context.Context, name string, conditions ...func() error) error {
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting release %#q", name))
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring deletion of release %#q", releaseName))
+		err := r.Delete(ctx, name)
+		if helmclient.IsReleaseNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q already deleted", name))
+		} else if err != nil {
+			return microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleted release %#q", name))
+		}
 
-	err := r.helmClient.DeleteRelease(releaseName, helm.DeletePurge(true))
-	if helmclient.IsReleaseNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q does not exist", releaseName))
-	} else if helmclient.IsTillerNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "warning", "message", "tiller is not found/installed")
-	} else if err != nil {
-		return microerror.Mask(err)
-	} else {
-		r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleted release %#q", releaseName))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured deletion of release %#q", name))
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured deletion of release %#q", releaseName))
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring conditions for deleted release %#q", name))
+
+		err := r.waitForConditions(ctx, conditions...)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured conditions for deleted release %#q", name))
+	}
 
 	return nil
 }
 
-func (r *Release) Install(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
-	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
+// EnsureInstalled makes sure the release is installed and all conditions are
+// met.
+//
+// NOTE: It does not update the release if it already exists.
+func (r *Release) EnsureInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating release %#q", name))
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating release %#q", releaseName))
+		err := r.Install(ctx, name, version, values)
```

Comment:
`conditions` aren't passed here on purpose. I added logging around them below. As stated in other comment I'd like to remove them soon.